### PR TITLE
chore(ci): disable continue-on-error for all test jobs in CI workflow

### PR DESCRIPTION
**Pull request by Qazalbash**
[`continue-on-error`](https://stackoverflow.com/a/73357322/26167473) was temporarily introduced in #1959 and accidentally got merged into the default branch. This PR reverts the commit (295136f12ffd185f110c58c669940ed0686224d8) introducing the temporary changes.

### CODE REVIEW

**Comment by Qazalbash** [comment]
@fehiepsi Some more test cases have failed!

Can you check `test/contrib/einstein/test_stein_loss.py::test_stein_particle_loss`? It has a relatively large error.

**Comment by fehiepsi** [comment]
@OlaRonning could you take a look? i think the inspected zs is different now.

**Comment by OlaRonning** [comment]
I think you're right @fehiepsi. I'll work it over in detail  in the morning.

File: test/contrib/einstein/test_stein_loss.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def stein_loss_fn(chosen_particle, obs, particles, assign):
     xs = jnp.array([-1, 0.5, 3.0])
     num_particles = xs.shape[0]
     particles = {"x": xs}
-    zs = jnp.array([-0.1241799, -0.65357316, -0.96147573])  # from inspect
+    zs = jnp.array([-3.3022664, -1.06049, 0.64527285])  # from inspect
```
Comment (fehiepsi) [inline test/contrib/einstein/test_stein_loss.py:83]: could we just replicate the logic to generate zs here? @OlaRonning

Comment (OlaRonning) [reply test/contrib/einstein/test_stein_loss.py:83]: Yeah, that should be possible. I'll have a look at it now.

**Comment by OlaRonning** [comment]
The stein test is passing but there is a problem with stochastic support.

**Comment by Qazalbash** [comment]
I fixed the very same problem for py3.10 in 78c51a702c4e0006dfb1bbc07cda7517277d4629, by changing the random seed, because the error was too high for the previous seed on which it was passing on py3.9. Now it is passing for py3.10 and not for py3.9. I presume we will encounter such cases in the future too! We have to find something more robust that will work on both Python versions.

**Comment by OlaRonning** [comment]
Looking at the failing assert in CI it looks like the dimensions are swapped. Could be spurious or could be that the inference in the test case is nonidentifiable. *Note: I haven't check the test or method in detail.*

**Comment by fehiepsi** [comment]
Yeah, we just need to assert whether the actual is close to expected or expected[::-1] (using np.is_close)

**Comment by OlaRonning** [comment]
This seems like the simpler solution than making the toy problem identifiable :+1:  I'll add it.

**Review by fehiepsi — APPROVED** [review]
Thanks @Qazalbash and @OlaRonning!
